utils/src/main/java/org/akvo/flow/utils/entity/Question.kt

```diff
@@ -22,34 +22,97 @@ package org.akvo.flow.utils.entity
 import java.util.HashMap
 
 data class Question(
-    val questionId: String? = null,
-    val isMandatory: Boolean = false,
+    var questionId: String? = null,
+    var isMandatory: Boolean = false,
     var text: String? = null,
-    val order: Int = 0,
+    var order: Int = 0,
     var isAllowOther: Boolean = false,
-    val renderType: String? = null,
+    var renderType: String? = null,
     var questionHelp: MutableList<QuestionHelp> = mutableListOf(),
-    val type: String? = null,
+    var type: String? = null,
     var options: MutableList<Option>? = null,
     var isAllowMultiple: Boolean = false,
-    val isLocked: Boolean = false,
-    val languageTranslationMap: HashMap<String?, AltText> = HashMap(),
-    val dependencies: MutableList<Dependency> = mutableListOf(),
-    val useStrength: Boolean = false,
-    val strengthMin: Int = 0,
-    val strengthMax: Int = 0,
-    val isLocaleName: Boolean = false,
-    val isLocaleLocation: Boolean = false,
-    val isDoubleEntry: Boolean = false,
-    val isAllowPoints: Boolean = false,
-    val isAllowLine: Boolean = false,
-    val isAllowPolygon: Boolean = false,
-    val caddisflyRes: String? = null,
-    val cascadeResource: String? = null,
-    val levels: MutableList<Level> = mutableListOf(),
+    var isLocked: Boolean = false,
+    var languageTranslationMap: HashMap<String?, AltText> = HashMap(),
+    var dependencies: MutableList<Dependency> = mutableListOf(),
+    var isLocaleName: Boolean = false,
+    var isLocaleLocation: Boolean = false,
+    var isDoubleEntry: Boolean = false,
+    var isAllowPoints: Boolean = false,
+    var isAllowLine: Boolean = false,
+    var isAllowPolygon: Boolean = false,
+    var caddisflyRes: String? = null,
+    var cascadeResource: String? = null,
+    var levels: MutableList<Level> = mutableListOf(),
+    var validationRule: ValidationRule? = null,
 ) {
 
     fun addAltText(altText: AltText) {
         languageTranslationMap[altText.languageCode] = altText
     }
+
+    //TODO: remove
```

Comment:
Why not remove now?

Reply:
I am working on this in part 6, it's not easy to just remove